components/ILIAS/Authentication/classes/Password/class.ilLocalPasswordSettingsGUI.php

```diff
@@ -0,0 +1,249 @@
+                    }
+
+                    break;
+                default:
+                    $title = $this->lng->txt('chg_ilias_password');
+
+                    break;
+            }
+            $section = $this->ui_factory->input()->field()->section($items, $title);
+            $items = ['password' => $section];
+        }
+
+        return $this->ui_factory->input()->container()->form()->standard(
+            $this->ctrl->getLinkTarget($this, 'savePassword'),
+            $items
+        )->withSubmitLabel($this->lng->txt('save'));
+    }
+
+    public function savePassword(): void
+    {
+        if (!$this->password_manager->allowPasswordChange($this->user)) {
+            $this->ctrl->redirect($this, 'showPersonalData');
+
+            return;
+        }
+
+        $form = $this->getPasswordForm()->withRequest($this->request);
+        $section = $form->getInputs()['password'];
+        /** @var Password $cp
+         * @var Password $np
+         */
+        $cp = $section->getInputs()['current_password'];
+        $np = $section->getInputs()['new_password'];
+        $errors = ['current_password' => [], 'new_password' => []];
+
+        $entered_current_password = $cp->getValue();
+        $entered_new_password = $np->getValue();
+
+        if (!$form->getError()) {
+            $data = $form->getData();
+            $error = false;
+
+            if ((int) $this->user->getAuthMode(true) === ilAuthUtils::AUTH_LOCAL) {
+                if (!ilLocalPasswordManager::getInstance()->verifyPassword(
+                    $this->user,
+                    $entered_current_password
+                )) {
+                    $error = true;
+                    $errors['current_password'][] = $this->lng->txt('passwd_wrong');
+                }
+            }
+
+            if (!ilSecuritySettingsChecker::isPassword($entered_new_password, $custom_error)) {
+                $error = true;
+                if ($custom_error !== '') {
+                    $errors['new_password'][] = $custom_error;
+                } else {
+                    $errors['new_password'][] = $this->lng->txt('passwd_invalid');
+                }
+            }
+            $error_lng_var = '';
+            if (!ilSecuritySettingsChecker::isPasswordValidForUserContext(
+                $entered_new_password,
+                $this->user,
+                $error_lng_var
+            )) {
+                $this->tpl->setOnScreenMessage('failure', $this->lng->txt('form_input_not_valid'));
+                $errors['new_password'][] = $this->lng->txt($error_lng_var);
+                $error = true;
+            }
+            if (
+                $entered_current_password === $entered_new_password &&
+                ($this->user->isPasswordExpired() || $this->user->isPasswordChangeDemanded())
+            ) {
+                $error = true;
+                $errors['new_password'][] = $this->lng->txt('new_pass_equals_old_pass');
+            }
```

Comment:
See my comment above, please check whether some code lines could be moved to transformations of the defined inputs.